tests/Unit/Domain/Test_Orientations.cpp

```diff
@@ -0,0 +1,118 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include <catch.hpp>
+
+#include "Domain/Direction.hpp"
+#include "Domain/Orientation.hpp"
+#include "Utilities/StdHelpers.hpp"
+
+namespace {
+
+void test_full_3d() {
+  // Test if FullOrientation can encode a 3D Flipped.
+  const auto& upper_xi = Direction<3>::upper_xi();
+  const auto& upper_eta = Direction<3>::upper_eta();
+  const auto& upper_zeta = Direction<3>::upper_zeta();
+  const auto& lower_xi = Direction<3>::lower_xi();
+  const auto& lower_eta = Direction<3>::lower_eta();
+  const auto& lower_zeta = Direction<3>::lower_zeta();
+
+  std::array<Direction<3>, 3> block_directions1{
+      {upper_xi, upper_eta, upper_zeta}};
+  std::array<Direction<3>, 3> block_directions2{
+      {lower_xi, lower_eta, lower_zeta}};
+  Orientation<3> custom_orientation(block_directions1, block_directions2);
+  CHECK(custom_orientation.mapped(upper_xi) == lower_xi);
+  CHECK(custom_orientation.mapped(upper_eta) == lower_eta);
+  CHECK(custom_orientation.mapped(upper_zeta) == lower_zeta);
+
+  Direction<3> direction(Direction<3>::Axis::Zeta, Side::Upper);
+  std::array<SegmentId, 3> segment_ids{
+      {SegmentId(2, 1), SegmentId(3, 1), SegmentId(3, 3)}};
+
+  std::array<SegmentId, 3> flipped_ids{
+      {SegmentId(2, 2), SegmentId(3, 6), SegmentId(3, 4)}};
+  // CHECK(!custom_orientation.isAligned());
+  CHECK(custom_orientation.mapped(2) == 2);
+  CHECK(custom_orientation.mapped(direction) == direction.opposite());
+  CHECK(custom_orientation.mapped(segment_ids) == flipped_ids);
+  CHECK(custom_orientation.is_aligned() == false);
+  Orientation<3> aligned_orientation(block_directions1, block_directions1);
+  CHECK(aligned_orientation.is_aligned() == true);
+}
+
+void test_full_2d() {
```

Comment:
2D test currently tests all possible right-handed 2D orientations, 3D only tests aligned and one non-aligned